manifests/broker/service.pp

```diff
@@ -12,38 +12,46 @@
   $service_ensure  = $kafka::broker::service_ensure,
   $jmx_opts        = $kafka::broker::jmx_opts,
   $log4j_opts      = $kafka::broker::log4j_opts,
-  $opts            = $kafka::broker::opts
+  $heap_opts       = $kafka::broker::heap_opts,
+  $opts            = $kafka::broker::opts,
 ) {
 
   if $caller_module_name != $module_name {
     fail("Use of private class ${name} by ${caller_module_name}")
   }
 
+  $service_name = 'kafka'
+
   if $service_install {
     if $::service_provider == 'systemd' {
       include ::systemd
 
-      file { '/usr/lib/systemd/system/kafka.service':
-        ensure  => present,
+      file { "${service_name}.service":
+        ensure  => file,
+        path    => "/usr/lib/systemd/system/${service_name}.service",
```

Comment:
overwriting anything in /usr/lib/systemd/system is a bad idea and not recommened. The prefered solution is to use /etc/systemd/system. The first path is meant to be used by package managers only. You can create a unit at the second path with an equal name which will kind of inherit from the first one. Also ubuntu or debian don't use /usr/lib but /lib/something.